setup-homebrew/main.sh

```diff
@@ -0,0 +1,49 @@
+#/bin/bash
+
+set -euo pipefail
+
+if which brew &>/dev/null; then
+    HOMEBREW_PREFIX="$(brew --prefix)"
+    HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX/Homebrew"
+    HOMEBREW_CORE_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core"
+    brew update-reset "$HOMEBREW_REPOSITORY" "$HOMEBREW_CORE_REPOSITORY"
+else
+    HOMEBREW_PREFIX=/home/linuxbrew/.linuxbrew
+    HOMEBREW_REPOSITORY="$HOMEBREW_PREFIX/Homebrew"
+    sudo mkdir -p "$HOMEBREW_PREFIX"
+    sudo git clone --depth=1 https://github.com/Homebrew/brew "$HOMEBREW_REPOSITORY"
+
+    HOMEBREW_CORE_REPOSITORY="$HOMEBREW_REPOSITORY/Library/Taps/homebrew/homebrew-core"
+    sudo mkdir -p "$HOMEBREW_CORE_REPOSITORY"
+    sudo rm -rf "$HOMEBREW_CORE_REPOSITORY"
+    sudo git clone --depth=1 https://github.com/Homebrew/linuxbrew-core "$HOMEBREW_CORE_REPOSITORY"
+
+    cd "$HOMEBREW_PREFIX"
+    sudo mkdir -p bin etc include lib opt sbin share var/homebrew/linked Cellar
+    sudo ln -sf ../Homebrew/bin/brew "$HOMEBREW_PREFIX/bin/"
+    cd -
+fi
+
+export PATH="$HOMEBREW_PREFIX/bin:$PATH"
+echo "::add-path::$HOMEBREW_PREFIX/bin"
+
+GEMS_PATH="$HOMEBREW_REPOSITORY/Library/Homebrew/vendor/bundle/ruby/"
+echo "::set-output name=gems-path::$GEMS_PATH"
+GEMS_HASH=$(shasum -a 256 "$HOMEBREW_REPOSITORY/Library/Homebrew/Gemfile.lock" | cut -f1 -d' ')
+echo "::set-output name=gems-hash::$GEMS_HASH"
+
+case "$GITHUB_REPOSITORY" in
+    */brew|*/*brew-core)
+        cd "$HOMEBREW_CORE_REPOSITORY"
+        rm -rf "$GITHUB_WORKSPACE"
+        ln -s "$HOMEBREW_CORE_REPOSITORY" "$GITHUB_WORKSPACE"
```

Comment:
Oops, this isn't quite right for Homebrew/brew; you want to link `HOMEBREW_REPOSITORY` in that case.

Reply:
And if you're adding a separate case: what you may want for other taps would be `ln -s "$HOMEBREW_REPOSITORY/Library/Taps/$GITHUB_REPOSITORY" "$GITHUB_WORKSPACE"` and then I believe it'll all work as expected.